rest/route/user.go

```diff
@@ -586,15 +586,19 @@ func removeHiddenProjects(permissions []rolemanager.PermissionSummary) error {
 type rolesPostRequest struct {
 	// the list of roles to add for the user
 	Roles []string `json:"roles"`
+	// The list of roles to remove for the user
+	RemoveRoles []string `json:"remove_roles"`
 	// if true, will also create a shell user document for the user. By default, specifying a user that does not exist will error
 	CreateUser bool `json:"create_user"`
 }
 
 type userRolesPostHandler struct {
-	rm         gimlet.RoleManager
-	userID     string
-	roles      []string
-	createUser bool
+	rm            gimlet.RoleManager
+	userID        string
+	caller        string
+	rolesToAdd    []string
+	rolesToRemove []string
```

Comment:
Should we prevent users from adding and removing the same role in one call? It's mostly harmless though so maybe not.

Reply:
I thought about that but yeah it felt like overkill, since it's not technically a problem, and also only we're using this